File: cgi-bin/add.py
import cgi, cgitb

form = cgi.FieldStorage()

a = int(form.getvalue('a'))
b = int(form.getvalue('b'))
result = a + b

print ("Content-type:text/html")
print()
print ("<html>")
print ('<head>')
print ("<title>Hello - Second CGI Program</title>")
print ('</head>')
print ('<body>')
print ("<h2>%i + %i = %i</h2>" % (a, b, result))
print ('</body>')
print ('</html>')

File: cgi-bin/add.php
<?php
$a = $_POST['a'];
$b = $_POST['b'];
$result = $a + $b;

echo "Content-type:text/html\n\n";
echo "<html>\n";
echo "<head>\n";
echo "<title>Hello - Second CGI Program</title>\n";
echo "</head>\n";
echo "<body>\n";
echo "<h2>$a + $b = $result</h2>\n";
echo "</body>\n";
echo "</html>";
?>
